internal/buildscripts/packaging/msi/splunk-otel-collector.wxs

```diff
@@ -43,14 +43,12 @@
                         <RegistryValue Type="string" Name="SPLUNK_CONFIG" Value="[CommonAppDataFolder]Splunk\OpenTelemetry Collector\agent_config.yaml"/>
                      </RegistryKey>
                   </Component>
+                  <Component Id="JmxMetricsJarComponent" Guid="c2aca15f-813a-4497-9bad-1c97e7bff618">
+                    <File Id="JmxMetricsJar" Name="opentelemetry-java-contrib-jmx-metrics.jar" KeyPath="yes" Source="$(var.JmxMetricsJar)"/>
```

Comment:
Are you introducing breaking changes and doc/usage requirements as the outcome of an attempt to reduce flake?

Reply:
I think it's also a matter of best practices on Windows. I'm not very familiar with paths on Windows and where files should be installed, so I'm up for correction here. It sounded like the program files directory where the other files installed by the `msi` was a more appropriate place.

This would be a breaking change for users though, you're correct. I'll have to add a note in the changelog if this is the accepted way to go.

The URL isn't actually documented anywhere that I've found. [Splunk's JMX receiver documentation](https://docs.splunk.com/observability/en/gdi/monitors-languages/jmx.html#jmx) is still pointing at the smart agent receiver. It's an ongoing effort to migrate the documentation, so this doesn't have a documentation impact since either path would need to be added to documentation.

Reply:
Ok, separating concerns here:

1. I think @pjanotti found the test failure cause, which is likely a real bug in our wix use where we aren't specifying the opt directory's drive and there's a race condition. Doing so would be the fix I'd advocate for, though I may not be tracking what's up here.
2. Windows conventions and the jmx receiver: this is an issue for the contrib component where the default path would be based on the detected goos/runtime. IMO as a distribution (wrapper) for the receiver we have an obligation to fix problems, conform to their implementation transparently, or if not possible/advisable, automatically follow required practices on the user's behalf (maybe w/ a config converter?). I think that's not needed here, or at least would have larger scope than warranted for the observed failures.

Reply:
1. @pjanotti would it be acceptable to you to keep the jmx at the same path, and find a way to determinately install and check a specific drive to resolve this? Or is it really bad practice to have the path be `Drive:\opt\`?
2. Yes, and for context, the receiver doesn't do any interesting logic for the path, it doesn't account for OS. So I agree this would be an issue to fix in `contrib`, not here.

Reply:
This is a "simple problem" which requires thinking carefully about the resolution... At first my preferential fix is to make the change in contrib, defaulting the env var according to `runtime.GOOS` and that should be a path based on `ProgramFiles`. Meanwhile, since that is going to take a bit of time, we can get rid of the test flakiness by updating the check (something using `Test-Path` on all drives on the box) - let me sketch such solution.

Reply:
Something like the code below should do it

```PowerShell
$found_jmx = $false
(Get-PSDrive -PSProvider FileSystem).Root | ForEach-Object {
    $found_jmx = $found_jmx -Or (Test-Path (Join-Path "$_" "opt\opentelemetry-java-contrib-jmx-metrics.jar"))
}
if (-not $found_jmx) { throw "Could not find expected jar file" }
```